amundsen_application/static/js/components/common/ResourceList/index.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import Pagination from 'react-js-pagination';
+import ResourceListItem from 'components/common/ResourceListItem';
+import { Resource } from 'interfaces';
+import { ITEMS_PER_PAGE, PAGINATION_PAGE_RANGE } from './constants';
+
+
+export interface ResourceListProps {
+  items: Resource[];
+  source: string;
+
+  // The following props are only used if 'paginate' is set to true
+  paginate?: boolean;
+  activePage?: number;
+  isFullList?: boolean;
+  itemsCount?: number;
+  itemsPerPage?: number;
+  onPagination?: (pageNumber: number) => void;
+}
+
+interface ResourceListState {
+  activePage: number;
+}
+
+class ResourceList extends React.Component<ResourceListProps, ResourceListState> {
+  public static defaultProps: Partial<ResourceListProps> = {
+    paginate: false,
+    activePage: 0,
+    itemsPerPage: ITEMS_PER_PAGE,
+    isFullList: true,
```

Comment:
Can you clarify the purpose of this prop? It looks like it's a boolean for whether or not the `items` list is the full list of items to show, but I got confused here:
```
 if (paginate && isFullList) { <-- full list yet we only show a portion?
      itemsToRender = items.slice(startIndex, startIndex + itemsPerPage);
 }
```

Is the `paginate` boolean not enough?

Reply:
There's actually a difference between `paginate` and `isFullList` which adds a bit to the complexity. I considered creating two separate components because of this.

For `MyBookmarks` the `items` prop is the full list of bookmarks. Displaying page 2 means we render bookmarks 5 - 8, so we would slice off the appropriate bookmarks from our full list and render them.

On the `SearchPage` the `items` prop is limited to 10 since that is what we get from the API response. To display page 2, we render all 10 `items` to render results 11 - 20 of X,XXX.

I guess the term `isFullList` is somewhat unclear. I also considered `sliceItems` or something similar.

Reply:
I'd go with `sliceItems`.